packages/jest-runner/__typetests__/jest-runner.test.ts

```diff
@@ -45,6 +48,32 @@ const callbackRunner = new CallbackRunner(globalConfig, runnerContext);
 expectType<boolean | undefined>(callbackRunner.isSerial);
 expectType<false>(callbackRunner.supportsEventEmitters);
 
+// CallbackTestRunnerInterface
+
+class CustomCallbackRunner implements CallbackTestRunnerInterface {
+  readonly #maxConcurrency: number;
+  readonly #globalConfig: Config.GlobalConfig;
+
+  constructor(globalConfig: Config.GlobalConfig) {
+    this.#globalConfig = globalConfig;
+    this.#maxConcurrency = globalConfig.maxWorkers;
+  }
```

Comment:
Note that `context` is omitted in constructor and members are named differently than in `extends CallbackTestRunner` example above. It is possible have `implements CallbackTestRunner` too, but then this sweet flexibility is not allowed. This code is from my actual use case. Learned something new.